contracts/Zauction.sol

```diff
@@ -93,33 +97,103 @@ contract ZAuction is Initializable, OwnableUpgradeable {
     }
     require(!consumed[bidder][auctionId], "zAuction: data already consumed");
 
-    // Will truncate any decimals
-    uint256 royalty = bid / 10;
+    uint256 rootId = rootDomainId[tokenId];
+    if (rootId == 0) {
+      rootId = rootDomainIdOf(tokenId);
+      rootDomainId[tokenId] = rootId;
+    }
 
-    IERC721 nftContract = IERC721(nftAddress);
     consumed[bidder][auctionId] = true;
-    // Bidder -> Owner, send funds
-    SafeERC20.safeTransferFrom(token, bidder, msg.sender, bid - royalty);
-    // Bidder -> Registrar, pay royalty
-    SafeERC20.safeTransferFrom(
-      token,
-      bidder,
-      registrar.minterOf(tokenId),
-      royalty
-    );
+
+    // Transfer payment and royalties
+    paymentTransfers(bidder, bid, msg.sender, rootId, tokenId);
+
     // Owner -> Bidder, send NFT
-    nftContract.safeTransferFrom(msg.sender, bidder, tokenId);
+    registrar.safeTransferFrom(msg.sender, bidder, tokenId);
+
     emit BidAccepted(
       auctionId,
       bidder,
       msg.sender,
       bid,
-      address(nftContract),
+      address(registrar),
       tokenId,
       expireBlock
     );
   }
 
+  function paymentTransfers(
+    address bidder,
+    uint256 bid,
+    address sender,
+    uint256 rootId,
+    uint256 tokenId
+  ) internal {
+    address rootOwner = registrar.ownerOf(rootId);
+    uint256 rootRoyalty = calculateRootOwnerRoyalty(rootId, bid, tokenId);
+    uint256 minterRoyalty = calculateMinterRoyalty(bid, tokenId);
+
+    uint256 bidActual = bid - minterRoyalty - rootRoyalty;
+
+    // Bidder -> Owner, pay transaction
+    SafeERC20.safeTransferFrom(token, bidder, sender, bidActual);
+
+    // Bidder -> Minter, pay minter royalty
+    SafeERC20.safeTransferFrom(
+      token,
+      bidder,
+      registrar.minterOf(tokenId),
+      minterRoyalty
+    );
+
+    // Bidder -> Root Owner, pay root owner royalty
+    SafeERC20.safeTransferFrom(token, bidder, rootOwner, rootRoyalty);
+  }
+
+  function calculateRootOwnerRoyalty(
+    uint256 rootId,
+    uint256 bid,
+    uint256 id
+  ) public view returns (uint256) {
+    require(id > 0, "zAuction: must provide a valid id");
+
+    // Find what percent they've specified as a royalty
+    uint256 royalty = rootDomainRoyalty[rootId];
+
+    // If not found or 0% royalty
+    if (royalty == 0) return 0;
+    // if not found because it's new do we just say 0?
+
+    // Pad with 18 zeroes for precision
+    uint256 divisor = (100 * 10**18) / (royalty * 10**18);
```

Comment:
Nit: One place in your code you use `(100 * (10**18))` and in this you use `(100 * 10**18)`, please make this consistent